forge/forge/agent/base.py

```diff
@@ -274,16 +271,21 @@ async def run_pipeline(
                 raise e
         return method_result
 
-    def serialize_configs(self) -> str:
+    class ModelContainer(BaseModel):
+        models: dict[str, BaseModel]
+
+    def dump_component_configs(self) -> str:
         configs = {}
         for component in self.components:
             if isinstance(component, ConfigurableComponent):
                 config_type_name = component.config.__class__.__name__
-                configs[config_type_name] = component.config.dict()
-        return json.dumps(configs, indent=4)
+                configs[config_type_name] = component.config
+        data = self.ModelContainer(models=configs).json()
+        raw = parse_raw_as(dict[str, dict[str, Any]], data)
+        return json.dumps(raw["models"], indent=4)
```

Comment:
Is this functionally different?
```suggestion
        return self.ModelContainer(models=configs).json(indent=4)
```

Reply:
Yes, your suggestion includes top level `"models"` key which I don't want because it's unnecessary.
I do all this because `self.ModelContainer(models=configs).json()` returns `Path` (or any other `json` unparsable type) to simple representation. Then I parse it to `dict` and can safely dump it as json while discarding `models` key.